File: Leetcode_solutions/Rotate Array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        for i in range(1, k + 1):
            now = nums.pop()
            nums.insert(0, now)
        return nums

    def rotate1(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        def reverse(nums, start, end):
            while start < end:
                temp = nums[start]
                nums[start] = nums[end]
                nums[end] = temp
                start += 1
                end -= 1
        k %= len(nums)
        reverse(nums, 0, len(nums) - 1)
        reverse(nums, 0, k - 1)
        reverse(nums, k, len(nums) - 1)

File: Leetcode_solutions/Rotate Array.java
class Solution {
    public void rotate(int[] nums, int k) {
        int temp, pre;
        for(int i=0; i<k; i++){
            pre = nums[nums.length-1];
            for(int j=0;j<nums.length;j++){
                temp=nums[j];
                nums[j]=pre;
                pre=temp;
            }
        }
    }
}